### PR TITLE
Lczaplinski/feature/rooming view

### PR DESCRIPTION
**Pull request by scoiatael**

### CODE REVIEW

File: rooms/serializers.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from django.shortcuts import reverse
+from django.forms.models import model_to_dict
+
+
+def room_to_dict(room):
+    model = model_to_dict(room)
+    model['join'] = reverse('rooms_join', kwargs={'room_id': room.pk})
+    model['is_locked'] = room.is_locked
+    model['free_places'] = room.capacity - room.users.count()
+    model.pop('users')
```
Comment (szewczykmira) [inline rooms/serializers.py:10]: Why do you pop users?

File: rooms/serializers.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    return model
+
+
+def user_to_dict(user):
+    name = user.username
+    if user.first_name or user.last_name:
+        name = "{} {}".format(user.first_name, user.last_name)
```
Comment (szewczykmira) [inline rooms/serializers.py:17]: What happen when `user.first_name` is None?

File: rooms/views.py [hunk 3]
```diff
@@ -43,45 +46,38 @@ def index(request):
...
     zosia = get_object_or_404(Zosia, active=True)
     can_start_rooming = zosia.can_start_rooming(get_object_or_404(UserPreferences, zosia=zosia, user=request.user))
-    rooms = Room.objects.for_zosia(zosia).all()
+    rooms = Room.objects.for_zosia(zosia).select_related('lock').prefetch_related('users').all()
     rooms_view = []
     for r in rooms:
-        d = model_to_dict(r)
-        d['free_places'] = r.free_places
-        d['is_locked'] = r.is_locked
+        d = room_to_dict(r)
```
Comment (szewczykmira) [inline rooms/views.py]: `d` and `r` is not good name for variable

File: zosia16/settings/dev.example.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
+from .common import *
+
+DEBUG = True
+# Set some predefined key to make development easier
+# SECRET_KEY = '...'
```
Comment (szewczykmira) [inline zosia16/settings/dev.example.py:5]: but we define `SECRET_KEY` in `common.py`

File: rooms/serializers.py (hunk 2 above)
Comment (scoiatael) [reply rooms/serializers.py:17]: "None Kowalsky" - I guess this should be `if user.first_name and user.last_name` ;)

File: rooms/serializers.py (hunk 1 above)
Comment (scoiatael) [reply rooms/serializers.py:10]: To avoid trying to serialize QuerySet

File: zosia16/settings/dev.example.py (hunk 4 above)
Comment (scoiatael) [reply zosia16/settings/dev.example.py:5]: But we generate random one there on app start, so if you override it here with constant one, you make sure it won't change.

File: rooms/views.py (hunk 3 above)
Comment (scoiatael) [reply rooms/views.py]: 👍

File: rooms/serializers.py [hunk 5]
```diff
@@ -13,7 +13,7 @@ def room_to_dict(room):
 
 def user_to_dict(user):
     name = user.username
-    if user.first_name or user.last_name:
+    if user.first_name and user.last_name:
         name = "{} {}".format(user.first_name, user.last_name)
```
Comment (szewczykmira) [inline rooms/serializers.py:17]: https://docs.djangoproject.com/en/1.10/topics/auth/customizing/#specifying-a-custom-user-model What about creating `get_full_name` method?

Comment (scoiatael) [reply rooms/serializers.py:17]: https://github.com/scoiatael/zosia16-site/issues/40 - I'll fix it in another PR.

**Review by szewczykmira — APPROVED** [review]

**Pull request merged**